content/compose/environment-variables/set-container-environment-variables.md

```diff
@@ -0,0 +1,103 @@
+
+### Additional information 
+- You can choose not to set a value and pass the environment variables from your shell straight through to your containers. It works in the same way as `docker run -e VARIABLE ...`:
+  ```yaml
+  web:
+    environment:
+      - DEBUG
+  ```
+  The value of the `DEBUG` variable in the container is taken from the value for the same variable in the shell in which Compose is run. 
+  Note that in this case no warning is issued if the `DEBUG` variable in the shell environment is not set. 
+
+- You can also take advantage of [interpolation](set-variables.md#interpolation-syntax).
+  ```yaml
+  web:
+    environment:
+      - DEBUG=${DEBUG}
+  ```
+  The result is similar to the one above but Compose gives you a warning if the `DEBUG` variable is not set in the shell environment.
+
+## Use the `env_file` attribute
+
+
+Container environment can also be set using [env files](set-variables.md#env-file) along with the [`env_file` attribute](../compose-file/05-services.md#env_file).
+
+```yaml
+services:
+  webapp:
+    env_file: "webapp.env"
+```
+
+The [`env_file` attribute](../compose-file/05-services.md#env_file) also lets you use multiple `.env` files in your Compose application. It can help you to keep your environment variables separate from your main configuration file, providing a more organized and secure way to manage sensitive information, as you do not need to place your `.env` file in the root of your project's directory. 
+
+Using an `.env` file for service environment allows you to use the same file for use by a plain `docker run --env-file ...` command,
+or to share same env file within multiple services without the need to duplicate a long `environment` yaml block.
+In addition, as the `.env` file supports [interpolation](set-variables.md#interpolation-syntax), it is possible to combine those with values set by `environment`. 
+
+> **Important**
+>
+> Interpolation in `.env` files is a Docker Compose CLI feature.
+>
+> It is not supported when running `docker run --env-file ...`.
+{ .important }
+
+### Additional information 
+- If multiple files are specified, they are evaluated in order and can override values set in previous files.
+- Environment variables declared in the `.env` file cannot then be referenced again separately in the Compose file.
+- If you use both the `env_file` and `environment` attribute, environment variables set by `environment` take precedence.
+- The paths to your `.env` file, specified in the `env_file` attribute,  are relative to the location of your `compose.yml` file. 
+- Values in your `.env` files can be overridden from the command line by using [`docker compose run -e`](set-variables.md#set-environment-variables-with-docker-compose-run---env).
+- Your `.env` files can be overriden by another `.env` if it is [substituted with `--env-file`](set-variables.md#substitute-with---env-file).
```

Comment:
`.env` is a file name, so there can't be "another `.env` ?
I'm not sure I understand what this recommendation is about